### DISCUSSION TITLE
Control of generated operationId name for operations

### DISCUSSION THREAD

**Discussion by Devon-White** [microsoft/typespec#4361] [OpenAPI3] [answered]
Is it possible to change the name that will be generated for a operation?

I have the following setup:

```css
@route("/resources/subscribers/{fabric_subscriber_id}/sip_endpoints")
namespace FabricAPI.SubscriberSipEndpoints {
  @tag("Subscriber SIP Endpoint")
  @friendlyName("SIP Endpoints")
  interface SIPEndpoints {
    @doc("A list of Sip Endpoints of the Subscriber")
    @friendlyName("List SIP Endpoints")
    list(...FabricSubscriberID):
      SubscriberSipEndpointListResponse |
      StatusCode401 |
      StatusCode404;
 ```
 
 This will generate as `SIPEndPoints_list`
 
 
 I would prefer for it to generate as `List SIP Endpoints`, just because it looks cleaner in documentation (we are currently generating the documentation off the OpenAPI spec)
 
 
 I expected `@friendlyName` to control this, but it doesn't seem too.
 
 Is this currently possible?

**Comment by Devon-White** [comment]
Here are some picture examples on how this looks like in Documentation (using Docusaurus):

Manually set (not using TypeSpec):
![image](https://github.com/user-attachments/assets/6e15b606-b79e-4075-85f3-578e41c4741a)


Using TypeSpec, and using the generated names:

![image](https://github.com/user-attachments/assets/cc2300c8-2e23-4252-977e-145d3f9f04a4)


Perhaps its a small personal preference, but I do think being able to control the the name of a operationID through `friendlyName` or a different decorator would be helpful.

**Comment by timotheeguerin** [comment, marked as answer]
You can use `@operationId` to be explicit about this value 
https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvb3BlbmFwaSI7Cgp1c2luZyBPcGVuQVBJOwoKCkBvcGVyYXRpb25JZCgiTXkgb3AgaWQiKQpvcCB0ZXN0KCk6IHZvaWQ7&e=%40typespec%2Fopenapi3&options=%7B%7D

I don't know if I have would have expected operationId to be used as a display name.
Does openapi `summary` property not get picked up here?

**Reply by Devon-White** [reply]
> You can use `@operationId` to be explicit about this value https://typespec.io/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvb3BlbmFwaSI7Cgp1c2luZyBPcGVuQVBJOwoKCkBvcGVyYXRpb25JZCgiTXkgb3AgaWQiKQpvcCB0ZXN0KCk6IHZvaWQ7&e=%40typespec%2Fopenapi3&options=%7B%7D
> 
> I don't know if I have would have expected operationId to be used as a display name. Does openapi `summary` property not get picked up here?

Great call outs. I found that if a `summary` is not present, it will use the `operationId` as the display name.

So using either the `operationID` or `summary` decorator achieves this behavior. Thanks for the assistance. :)
